adds "Add signature" feature to issue close page, ie if you send issue close
comments to all issue notification list members it's nice to have email
signature appended. feature request by slay.

--- eventum-1.7.1/close.php	2006-03-31 12:03:17.580792647 +0300
+++ /home/glen/close.php	2006-03-31 12:04:45.222748601 +0300
@@ -80,5 +80,9 @@
     ));
 }
 
+$usr_id = Auth::getUserID();
+$user_prefs = Prefs::get($usr_id);
+$tpl->assign("current_user_prefs", $user_prefs);
+
 $tpl->displayTemplate();
-?>
\ No newline at end of file
+?>
--- eventum-1.7.1/templates/en/close.tpl.html	2006-03-31 12:03:17.310786621 +0300
+++ /home/glen/close.tpl.html	2006-03-31 12:04:47.462798592 +0300
@@ -90,6 +90,22 @@
         cell.style.background = "{/literal}{$internal_color}{literal}";
     }
 }
+
+var old_reason = '';
+function setSignature(f)
+{
+{/literal}
+    var signature = "{$current_user_prefs.email_signature|replace:'"':'\"'|replace:"\r":""|replace:"\n":'\n'}";
+{literal}
+    if (f.add_email_signature.checked) {
+        old_reason = f.reason.value;
+        f.reason.value += "\n";
+        f.reason.value += signature;
+    } else {
+        f.reason.value = old_reason;
+    }
+}
+
 //-->
 </script>
 {/literal}
@@ -159,7 +175,10 @@
             <b>Reason for closing issue: *</b><br />
           </td>
           <td bgcolor="{$light_color}" class="default">
-            <textarea name="reason" rows="8" style="width: 97%"></textarea>
+            <textarea name="reason" rows="8" style="width: 97%">{if $current_user_prefs.auto_append_sig == 'yes'}
+
+
+{$current_user_prefs.email_signature|escape:"html"}{/if}</textarea>
             {include file="error_icon.tpl.html" field="reason"}
           </td>
         </tr>
@@ -205,6 +224,16 @@
               <tr>
                 <td><input class="button" type="button" value="&lt;&lt; Back" onClick="javascript:history.go(-1);"></td>
                 <td width="100%" align="center"><input class="button" type="submit" value="Close Issue"></td>
+
+                {if $current_user_prefs.email_signature != "" and $current_user_prefs.auto_append_sig != 'yes'}
+                <td class="default_white" align="right" width="150">
+                  <nobr>
+                  <input type="checkbox" name="add_email_signature" value="yes" onClick="javascript:setSignature(this.form);">
+                  <a id="white_link" class="white_link" href="javascript:void(null);" onClick="javascript:toggleCheckbox('close_form', 'add_email_signature');setSignature(getForm('close_form'));">Add Email Signature</a>
+                  &nbsp;&nbsp;
+                  </nobr>
+                </td>
+                {/if}
               </tr>
             </table>
           </td>
